src/schema/document/value.rs

```diff
@@ -261,9 +260,9 @@ impl<'a> ReferenceValueLeaf<'a> {
     #[inline]
     /// If the Value is a pre-tokenized string, returns the associated string. Returns None
     /// otherwise.
-    pub fn as_pre_tokenized_text(&self) -> Option<&'a PreTokenizedString> {
+    pub fn as_pre_tokenized_text(&self) -> Option<Box<PreTokenizedString>> {
```

Comment:
This could still return `Option<&PreTokenizedString>` even if internally, the storage is owned, to avoid the unconditional `clone` in the accessor. (The caller can `clone` or `cloned` this just as well.)

Reply:
so where would the object live

Reply:
Note, that it cannot return `&'a PreTokenizedString`, but it can still return `&'self PreTokenizedString`, i.e. a short-lived reference into the `Box` bound to the `&self` reference, which the caller can clone if so desired.